### PR TITLE
Hystrix metadata added to tags and Command Group included in resource name

### PR DESCRIPTION
**Pull request by blockmar**
Related to issue #707 

Changes:
- Added hystrix `Command Group` to resource name for a better grouping of commands.
- Added `Command Key`, `Command Group` and "is Circuit Breaker Open" as tags.

Example of resource name change:

Old name: `ExampleCommand.run`
New name: `ExampleGroup.ExampleCommand.run`

### CODE REVIEW

File: dd-java-agent/instrumentation/hystrix-1.4/hystrix-1.4.gradle [hunk 1]
```diff
@@ -17,7 +17,7 @@ testSets {
 }
 
 dependencies {
-//  compileOnly group: 'com.netflix.hystrix', name: 'hystrix-core', version: '1.5.12'
+  compileOnly group: 'com.netflix.hystrix', name: 'hystrix-core', version: '1.4.0'
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/hystrix-1.4/hystrix-1.4.gradle]: Nice catch...

File: dd-java-agent/instrumentation/hystrix-1.4/src/main/java/datadog/trace/instrumentation/hystrix/HystrixCommandInstrumentation.java [hunk 2]
```diff
@@ -48,24 +49,21 @@ public HystrixCommandInstrumentation() {
...
-        className = declaringClass.getName();
-        if (declaringClass.getPackage() != null) {
-          final String pkgName = declaringClass.getPackage().getName();
-          if (!pkgName.isEmpty()) {
-            className = declaringClass.getName().replace(pkgName, "").substring(1);
-          }
-        }
-      }
-      final String resourceName = className + "." + method.getName();
+    public static Scope startSpan(@Advice.This final HystrixCommand<?> command, @Advice.Origin final Method method) {
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/hystrix-1.4/src/main/java/datadog/trace/instrumentation/hystrix/HystrixCommandInstrumentation.java]: I'm guessing this line is too long which is why the formatter check isn't happy.  You can run `./gradlew googleJavaFormat` to have it reformat automatically.  (I think that's the right command.)

Comment (tylerbenson) [inline dd-java-agent/instrumentation/hystrix-1.4/src/main/java/datadog/trace/instrumentation/hystrix/HystrixCommandInstrumentation.java]: Could you change method to 
```java
@Advice.Origin("#m") final String methodName
```

**Comment by blockmar** [comment]
@tylerbenson Please take another look.

**Comment by blockmar** [comment]
@tylerbenson I'm not sure how to build the agent jar but if you could direct me to some instructions I would be happy to test it live before merging.

**Comment by tylerbenson** [comment]
@blockmar you can get the jar from the circleci `build` job.  It's saved as an artifact: https://18608-89221572-gh.circle-artifacts.com/0/home/circleci/dd-trace-java/libs/dd-java-agent-0.24.0-SNAPSHOT.jar

**Comment by blockmar** [comment]
@tylerbenson Now tested and it works. Unfortunately https://github.com/DataDog/dd-trace-java/issues/708 still makes the added value very low since tags are not exposed for Monitors.

But the improved resource name helps with insights.

**Comment by tylerbenson** [comment]
Great, thanks for confirming.  I have a different idea on how to address the service name problem...  I think it should be a general configuration for all integrations to allow defaults to be overridden.

**Pull request merged**